### PR TITLE
Add Use-Cases to Developer Guide (DG)

### PR DESCRIPTION
**Pull request by jamesyeap**
Adds Use-Cases for the following features to the Developer Guide (DG),
1. Add a temporary class
2. Add a recurring class
3. Delete a class
4. Assign a student to a class
5. Delete a student

Additionally, this pull-request removes placeholder details from the Developer Guide (DG).

Also, it updates the placeholder information in the Index page with the actual details of our project.

### CODE REVIEW

**Comment by jamesyeap** [comment]
Fixes #25

File: docs/index.md [hunk 1]
```diff
@@ -8,10 +8,11 @@ title: TeachWhat!
 
 ![Ui](images/Ui.png)
 
-**AddressBook is a desktop application for managing your contact details.** While it has a GUI, most of the user interactions happen using a CLI (Command Line Interface).
+**TeachWhat! is a desktop application used for private tutors in managing their class.** The user interacts
+with it using a CLI, and it has a GUI created with JavaFX.
 
-* If you are interested in using AddressBook, head over to the [_Quick Start_ section of the **User Guide**](UserGuide.html#quick-start).
-* If you are interested about developing AddressBook, the [**Developer Guide**](DeveloperGuide.html) is a good place to start.
+* If you are interested in using TeachWhat, head over to the [_Quick Start_ section of the **User Guide**](UserGuide.html#quick-start).
```
Comment (nnmoq) [inline docs/index.md:14]:
```suggestion
* If you are interested in using TeachWhat!, head over to the [_Quick Start_ section of the **User Guide**](UserGuide.html#quick-start).
```

File: docs/index.md [hunk 2]
```diff
@@ -8,10 +8,11 @@ title: TeachWhat!
...
+* If you are interested about developing TeachWhat, the [**Developer Guide**](DeveloperGuide.html) is a good place to start.
```
Comment (nnmoq) [inline docs/index.md:15]:
```suggestion
* If you are interested about developing TeachWhat!, the [**Developer Guide**](DeveloperGuide.html) is a good place to start.
```

**Review by nnmoq — CHANGES_REQUESTED** [review]
The application name is wrong.

**Review by nnmoq — APPROVED** [review]

**Pull request merged**